lib/kino_vega_lite/chart_cell.ex

```diff
@@ -168,18 +179,39 @@ defmodule KinoVegaLite.ChartCell do
     %{
       "chart_type" => "point",
       "data_variable" => value,
-      "x_field" => x_field,
-      "y_field" => y_field,
+      "x_field" => x_field.name,
+      "y_field" => y_field.name,
       "color_field" => nil,
-      "x_field_type" => nil,
-      "y_field_type" => nil,
+      "x_field_type" => x_field.type,
+      "y_field_type" => y_field.type,
       "color_field_type" => nil,
       "x_field_aggregate" => nil,
       "y_field_aggregate" => nil,
       "color_field_aggregate" => nil
     }
   end
 
+  defp updates_for_field_with_types(ctx, field, idx, value) do
+    layer = get_in(ctx.assigns.layers, [Access.at(idx)])
+
+    columns =
+      Enum.find_value(
+        ctx.assigns.data_options,
+        [],
+        &(&1.variable == layer["data_variable"] && &1.columns)
+      )
+
+    type = get_in(columns, [Access.filter(&(&1.name == value)), :type]) |> List.first()
```

Comment:
```suggestion
    %{type: type} = Enum.find(columns, &(&1.name == value))
```

Reply:
Only here I've solved it a little differently coz `type` can be `nil`